.github/workflows/python-package.yml

```diff
@@ -12,22 +12,23 @@ on:
 jobs:
   pytest:
     strategy:
+      fail-fast: false
       matrix:
         platform: [ ubuntu-latest, macos-latest, windows-latest ]
-        python-version: [ "3.7", "3.8", "3.9", "3.10", "3.11" ]
+        python-version: [ "3.7", "3.8", "3.9", "3.10", "3.11", "3.12" ]
```

Comment:
Even if this is not part of your PR, and since there is a build error with `macos-latest` and `3.7`, can you exclude it and add:

```yaml
exclude:
  - platform: macos-latest
    python-version: "3.7"
```